lib/main.js

```diff
@@ -190,6 +190,23 @@ class JavaLanguageClient extends AutoLanguageClient {
     connection.onCustom('language/actionableNotification', this.actionableNotification.bind(this))
   }
 
+  getInitializeParams(projectPath, process) {
+    const params = super.getInitializeParams(projectPath, process);
+    if (!params.initializationOptions) {
+      params.initializationOptions = {};
+    }
+    params.initializationOptions.bundles = this.collectJavaExtensions();
+    return params;
+  }
+
+  collectJavaExtensions() {
+    const allExtensions = [];
+    atom.packages.getLoadedPackages()
+        .filter(pkg => Array.isArray(pkg.metadata.javaExtensions))
+        .forEach(pkg => allExtensions.push(...pkg.metadata.javaExtensions.map(p => path.resolve(pkg.path, p))));
```

Comment:
Feels weird to build a new collection from inside a forEach... In C# I would have just done a SelectMany here and returned the collection that way.

Reply:
It does :-) but there is no `flatMap()` in JS6... would have to improvise with `map()` and `reduce()`

Reply:
This is how it'd look like:
```
  collectJavaExtensions() {
    return atom.packages.getLoadedPackages()
        .filter(pkg => Array.isArray(pkg.metadata.javaExtensions))
        .map(pkg => pkg.metadata.javaExtensions.map(p => path.resolve(pkg.path, p)))
        .reduce(e => e.concat([]));
  }
```

Reply:
That looks good - functional FTW :)